website/project/model.py

```diff
@@ -1128,6 +1146,36 @@ def can_comment(self, auth):
             )
         return self.is_contributor(auth.user)
 
+    def set_node_license(self, data, auth, save=True):
```

Comment:
Also, it might make sense to pass `id`, `year`, and `copyright_holders` instead of `data`. This makes it clear what the expected inputs are.